### ISSUE TITLE
Building with -DSEXP_USE_STATIC_LIBS fails

### ISSUE THREAD

**Issue by lambdadog** [ashinn/chibi-scheme#566]
Running 

```
$ make clibs.c
$ make CPPFLAGS=-DSEXP_USE_STATIC_LIBS
```

fails with

```
cc -DSEXP_USE_STATIC_LIBS -Iinclude  -Wall -g -g3 -O3   -o chibi-scheme main.o -L. -lchibi-scheme
/usr/bin/ld: ./libchibi-scheme.so: undefined reference to `sexp_static_libraries'
collect2: error: ld returned 1 exit status
make: *** [Makefile:171: chibi-scheme] Error 1
```

This is also the case for when using it to build statically (my usecase):

```
$ make clibs.c
$ make -B chibi-scheme-static SEXP_USE_DL=0 CFLAGS=-DSEXP_USE_STATIC_LIBS
...
cc -Wall -DSEXP_USE_DL=0 -g -g3 -O3  -static -DSEXP_USE_DL=0 -o chibi-scheme-static main.o gc.o sexp.o bignum.o gc_heap.o opcodes.o vm.o eval.o simplify.o   -lm
/usr/bin/ld: eval.o: in function `sexp_find_static_library':
/home/pea/Downloads/chibi-scheme-0.8.0/eval.c:1317: undefined reference to `sexp_static_libraries'
/usr/bin/ld: /home/pea/Downloads/chibi-scheme-0.8.0/eval.c:1317: undefined reference to `sexp_static_libraries'
collect2: error: ld returned 1 exit status
make: *** [Makefile:174: chibi-scheme-static] Error 1
```

**Comment by ashinn** [comment]
Currently the build tests only check for this in combination with the following flags:

`CFLAGS=-DSEXP_USE_STATIC_LIBS_NO_INCLUDE=0;CPPFLAGS=-DSEXP_USE_STATIC_LIBS=1`

(This is also broken recently since (chibi pty) needs `-lutil`.)

**Comment by ashinn** [comment]
Fixed in abc3403e0aed2932c0e7479070d48c5d9182bf32.
